wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/Command.java

```diff
@@ -200,6 +200,58 @@ public ParallelRaceGroup until(BooleanSupplier condition) {
     return raceWith(new WaitUntilCommand(condition));
   }
 
+  /**
+   * Decorates this command with a condition to wait for after this command is finished. The command will finish after
+   * this condition is met.
+   *
+   * <p>Note: This decorator works by adding this command to a composition. The command the
+   * decorator was called on cannot be scheduled independently or be added to a different
+   * composition (namely, decorators), unless it is manually cleared from the list of composed
+   * commands with {@link CommandScheduler#removeComposedCommand(Command)}. The command composition
+   * returned from this method can be further decorated without issue.
+   *
+   * @param condition the condition to wait for after this command is finished
+   * @return the command with the condition to await
+   */
+  public SequentialCommandGroup thenAwait(BooleanSupplier condition) {
+    return andThen(Commands.waitUntil(condition));
+  }
+
+    /**
+   * Decorates this command with a timeout to pass for after this command is finished. The command will finish after
+   * this timeout has passed.
+   *
+   * <p>Note: This decorator works by adding this command to a composition. The command the
+   * decorator was called on cannot be scheduled independently or be added to a different
+   * composition (namely, decorators), unless it is manually cleared from the list of composed
+   * commands with {@link CommandScheduler#removeComposedCommand(Command)}. The command composition
+   * returned from this method can be further decorated without issue.
+   *
+   * @param seconds the timeout duration
+   * @return the command with the timeout to await
+   */
+  public SequentialCommandGroup thenAwaitTimeout(double seconds) {
```

Comment:
```suggestion
  public SequentialCommandGroup thenTimeout(double seconds) {
```

Reply:
I think this might be a little less clear? Because you don't know what the timeout is there for.
But I think my choice of "timeout" is not very good, other suggestions are welcome

Reply:
I think this might be a little less clear? Because you don't know what the timeout is there for.
But I think my choice of "timeout" is not very good, other suggestions are welcome

Reply:
`thenWait`?

Reply:
Yes I think that's better! I'll change it tomorrow

Reply:
Actually I wonder if it's not too similar to `thenAwait`.